packages/atomic/src/components/ipx/atomic-ipx-refine-toggle/atomic-ipx-refine-toggle.tsx

```diff
@@ -0,0 +1,106 @@
+import {
+  BreadcrumbManager,
+  BreadcrumbManagerState,
+  buildBreadcrumbManager,
+  buildSearchStatus,
+  SearchStatus,
+  SearchStatusState,
+} from '@coveo/headless';
+import {Component, h, State, Element} from '@stencil/core';
+import FilterIcon from '../../../images/filter.svg';
+import {
+  InitializeBindings,
+  BindStateToController,
+} from '../../../utils/initialization-utils';
+import {Bindings} from '../../search/atomic-search-interface/atomic-search-interface';
+
+/**
+ * @internal
+ */
+@Component({
+  tag: 'atomic-ipx-refine-toggle',
+  styleUrl: 'atomic-ipx-refine-toggle.pcss',
+  shadow: true,
+})
+export class AtomicIPXRefineToggle {
+  @InitializeBindings() public bindings!: Bindings;
+  @Element() public host!: HTMLElement;
+
+  @State() public error!: Error;
+
+  @BindStateToController('breadcrumbManager')
+  @State()
+  private breadcrumbManagerState!: BreadcrumbManagerState;
+
+  @BindStateToController('searchStatus')
+  @State()
+  private searchStatusState!: SearchStatusState;
+
+  public breadcrumbManager!: BreadcrumbManager;
+  public searchStatus!: SearchStatus;
+  private modalRef?: HTMLAtomicIpxRefineModalElement;
+  private buttonRef?: HTMLButtonElement;
+
+  private get numberOfBreadcrumbs(): number {
```

Comment:
We ought to put this value `numberOfBreadcrumbs` in the Headless BreadcrumbManagerState perhaps...

Reply:
We could yeah, I can move it there in this PR if you want

Reply:
Alright, I'll open a jira (in your backlog) that I'll link in a comment